tests/director/tests/formation_mapping_api_test.go

```diff
@@ -0,0 +1,558 @@
+)
+
+type RequestBody struct {
+	State         ConfigurationState `json:"state"`
+	Configuration json.RawMessage    `json:"configuration"`
+	Error         string             `json:"error"`
+}
+
+type ConfigurationState string
+
+const (
+	ReadyConfigurationState         ConfigurationState = "READY"
+	CreateErrorConfigurationState   ConfigurationState = "CREATE_ERROR"
+	DeleteErrorConfigurationState   ConfigurationState = "DELETE_ERROR"
+	ConfigPendingConfigurationState ConfigurationState = "CONFIG_PENDING"
+
+	formationIDPathParam           = "ucl-formation-id"
+	formationAssignmentIDPathParam = "ucl-assignment-id"
+)
+
+func Test_UpdateStatus(t *testing.T) {
+	ctx := context.Background()
+	parentTenantID := tenant.TestTenants.GetDefaultTenantID()
+	subaccountID := tenant.TestTenants.GetIDByName(t, tenant.TestProviderSubaccount)
+
+	intSysName := "async-formation-int-system"
+	testConfig := `{"testCfgKey":"testCfgValue"}`
+	certSecuredHTTPClient := gql.NewCertAuthorizedHTTPClient(cc.Get()[conf.ExternalClientCertSecretName].PrivateKey, cc.Get()[conf.ExternalClientCertSecretName].Certificate, conf.SkipSSLValidation)
+
+	t.Run("Caller successfully updates formation assignment for himself", func(t *testing.T) {
+		runtimeInput := graphql.RuntimeRegisterInput{
+			Name:        "selfRegisterRuntimeAsync",
+			Description: ptr.String("selfRegisterRuntimeAsync-description"),
+		}
+
+		t.Log("Register runtime via certificate secured client")
+		runtime := fixtures.RegisterKymaRuntime(t, ctx, certSecuredGraphQLClient, subaccountID, runtimeInput, conf.GatewayOauth)
+		defer fixtures.CleanupRuntime(t, ctx, certSecuredGraphQLClient, subaccountID, &runtime)
+
+		appName := "testAsyncApp"
+		appType := "async-app-type-1"
+		t.Logf("Register application with name: %q", appName)
+		app, err := fixtures.RegisterApplicationWithApplicationType(t, ctx, certSecuredGraphQLClient, appName, conf.ApplicationTypeLabelKey, appType, parentTenantID)
+		defer fixtures.CleanupApplication(t, ctx, certSecuredGraphQLClient, parentTenantID, &app)
+		require.NoError(t, err)
+		require.NotEmpty(t, app.ID)
+
+		asyncFormationTmplName := "async-formation-template-name"
+		t.Logf("Creating formation template with name: %q", asyncFormationTmplName)
+		ft := createFormationTemplate(t, ctx, asyncFormationTmplName, conf.KymaRuntimeTypeLabelValue, []string{appType}, graphql.ArtifactTypeEnvironmentInstance)
+		defer fixtures.CleanupFormationTemplate(t, ctx, certSecuredGraphQLClient, ft.ID)
+
+		asyncFormationName := "async-formation-name"
+		t.Logf("Creating formation with name: %q from template with name: %q", asyncFormationName, asyncFormationTmplName)
+		formation := fixtures.CreateFormationFromTemplateWithinTenant(t, ctx, certSecuredGraphQLClient, parentTenantID, asyncFormationName, &asyncFormationTmplName)
+		defer fixtures.DeleteFormationWithinTenant(t, ctx, certSecuredGraphQLClient, parentTenantID, asyncFormationName)
+		formationID := formation.ID
```

Comment:
`require.NotNil(formation)` just in case, if you decide to add them in the previous two places.

Reply:
I'm not sure we need this because if you look at the `CreateFormationFromTemplateWithinTenant` function there is `require.Equal(t, formationName, formation.Name)` check and at the mentioned point I believe we cannot receive nil value. Even if there is a problem and nil value it will fail in the `require.Equal(t, formationName, formation.Name)`. But I think that's the reason of `require.NoError(t, err)` check, to validate there is no error hence formation is not nil?